### PR TITLE
Debug builds

### PR DESCRIPTION
**Pull request by Sriep**
Outlines how to prepare the OS for 0chian builds outside docker containers and suggests how to set up for debugging running 0chains.

### CODE REVIEW

File: local/config/0chain.yaml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      window: 100000 #number of blocks, Do not make 0 with minio ON, Should be less than minio old block round range
+      repeat_interval_mins: 1 #minutes
+      report_status_mins: 1 #minutes
+      batch_size: 50
+  lfb_ticket:
+    rebroadcast_timeout: "15s"  #
+    ahead: 5                    # should be >= 5
+    fb_fetching_lifetime: "10s" #
+  async_blocks_fetching:
+    max_simultaneous_from_miners: 100S
```
Comment (kirillt) [inline local/config/0chain.yaml:108]: is `S` a typo here?

File: local/config/b0magicBlock_4_miners_1_sharder.tmp.json [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  "starting_round": 0,
+  "miners": {
+    "type": 0,
+    "nodes": {
+      "31810bd1258ae95955fb40c7ef72498a556d3587121376d9059119d280f34929": {
+        "id": "31810bd1258ae95955fb40c7ef72498a556d3587121376d9059119d280f34929",
+        "version": "",
+        "creation_date": 0,
+        "public_key": "255452b9f49ebb8c8b8fcec9f0bd8a4284e540be1286bd562578e7e59765e41a7aada04c9e2ad3e28f79aacb0f1be66715535a87983843fea81f23d8011e728b",
+        "n2n_host": <Enter
```
Comment (kirillt) [inline local/config/b0magicBlock_4_miners_1_sharder.tmp.json:14]: Is this placeholder intended to be here or forgotten to change?

File: local/dubug_environment.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# Debug environment for 0Chain go executables
```
Comment (kirillt) [inline local/dubug_environment.md:1]: This doc is really valuable, I'll look at it closer little bit later.

**Review by kirillt — COMMENTED** [review]
Basically, this looks to be ready to be merged.

But please, squash commits into more concise history:
we have more than 10 commits here and half of them has not really informative messages,
with `git rebase -i 52b92356ff` you can squash several contiguous commits into one and also change messages if you want.

File: local/config/b0magicBlock_4_miners_1_sharder.tmp.json (hunk 2 above)
Comment (Sriep) [reply local/config/b0magicBlock_4_miners_1_sharder.tmp.json:14]: Place holder for you to put in your own IP addresses

File: local/config/0chain.yaml (hunk 1 above)
Comment (Sriep) [reply local/config/0chain.yaml:108]: Yes. I only changed the magic_block_file entry.

File: local/build_environment.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+env_posix.cc:(.text+0xf0): undefined reference to `dlclose'
+
+```
+Suggests a linker error, probably a problem with your RocksDB and gcc versions.
+```shell
+/usr/bin/ld: /usr/local/lib/librocksdb.a(format.o): in function `rocksdb::LZ4_Uncompress(rocksdb::UncompressionContext const&, char const*, unsigned long, int*, unsigned int, rocksdb::MemoryAllocator*)':
+format.cc:(.text._ZN7rocksdb14LZ4_UncompressERKNS_20UncompressionContextEPKcmPijPNS_15MemoryAllocatorE[_ZN7rocksdb14LZ4_UncompressERKNS_20UncompressionContextEPKcmPijPNS_15MemoryAllocatorE]+0xd5): undefined reference to `LZ4_createStreamDecode'
+/usr/bin/ld: format.cc:(.text._ZN7rocksdb14LZ4_UncompressERKNS_20UncompressionContextEPKcmPijPNS_15MemoryAllocatorE[_ZN7rocksdb14LZ4_UncompressERKNS_20UncompressionContextEPKcmPijPNS_15MemoryAllocatorE]+0x135): undefined reference to `LZ4_setStreamDecode'
+```
+TODO: Still working on this.
```
[the remark this answers is not in the document]
Comment (Sriep) [reply local/build_environment.md:151]: I got stuck on it, so I fixed the issue on the machine I was having it on, but copying the libraries from another machine. Contemplate what to do about the todo.
The second is a warning that had not apparent bad effects.

File: local/dubug_environment.md (hunk 3 above)
Comment (Sriep) [reply local/dubug_environment.md:1]: I need to redo the screen prints, plus add a start debugging checklist.

File: local/bin/clean.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+rm -r "$(dirname "$0")"/../miner
+rm -r "$(dirname "$0")"/../sharder
```
[the remark this answers is not in the document]
Comment (Sriep) [reply local/bin/clean.sh:4]: Wow! You lean something every day!

File: local/config/b0magicBlock_4_miners_1_sharder.tmp.json (hunk 2 above)
Comment (kirillt) [reply local/config/b0magicBlock_4_miners_1_sharder.tmp.json]: That's fine actually, good approach for making a user aware of placeholder.
Maybe only would be good to make it single-line, but not so important.
Just wanted to make sure that it isn't something accidental.

**Pull request merged**